skimage/draw/draw3d.py

```diff
@@ -63,6 +64,165 @@ def ellipsoid(a, b, c, spacing=(1., 1., 1.), levelset=False):
+                        \sin\theta &  \cos\theta & 0 \\
+                                 0 &           0 & 1 \\
+                    \end{bmatrix}
+
+        R(\psi) = \begin{bmatrix}
+                      1 &        0 &         0 \\
+                      0 & \cos\psi & -\sin\psi \\
+                      0 & \sin\psi &  \cos\psi \\
+                           \end{bmatrix}
+
+        \\
+
+        R(\phi, \theta, \psi) = (R(\psi)R(\theta)R(\phi))^\intercal
+
+    """
+
+    if len(center) != 3:
+        raise ValueError(f'len(center) should be 3 but got {len(center)}')
+    center = np.array(center)
+
+    if len(axis_lengths) != 3:
+        raise ValueError(
+            f'len(axis_lengths) should be 3 but got {len(axis_lengths)}')
+    axis_lengths = np.array(axis_lengths)
+
+    if angles is None:
+        R = np.eye(3)
+    else:
+        if len(angles) != 3:
+            raise ValueError(
+                'len(angles) should be 3 but got',
+                len(angles))
+        if axes is None:
+            axes = range(3)
+        if (len(axes) != 3
+            or not set(axes).issubset(range(3))
+            or axes[0] == axes[1]
+                or axes[1] == axes[2]):
+            raise ValueError(f'axes: {axes} is invalid')
+
+        axes_str = 'ZYX' if intrinsic else 'zyx'
+        seq = ''.join(axes_str[axis] for axis in axes)
+
+        # Generate a rotation matrix. The order of the elements needs to be
+        # reversed so that it follows the (z, y, x) order.
+        R = Rotation.from_euler(seq, angles).as_matrix()[::-1, ::-1]
+
+    if spacing is None:
+        spacing = np.ones(3)
+    elif len(spacing) != 3:
+        raise ValueError(f'len(spacing) should be 3 but got {len(spacing)}')
+    spacing = np.array(spacing)
+    scaled_center = center / spacing
+
+    # The upper_left_bottom and lower_right_top corners of the smallest cuboid
+    # containing the ellipsoid.
+    factor = np.array([
+        [i, j, k] for k in (-1, 1) for j in (-1, 1) for i in (-1, 1)
+    ]).T
+    axis_lengths_rot = np.abs(
+        np.diag(1. / spacing) @ (R @ (np.diag(axis_lengths) @ factor))
+    ).max(axis=1)
+    upper_left_bottom = np.ceil(scaled_center - axis_lengths_rot).astype(int)
+    lower_right_top = np.floor(scaled_center + axis_lengths_rot).astype(int)
+
+    if shape is not None:
+        # Constrain upper_left and lower_ight by shape boundary.
+        upper_left_bottom = np.maximum(upper_left_bottom, np.array([0, 0, 0]))
+        lower_right_top = np.minimum(lower_right_top, np.array(shape[:3]) - 1)
+
+    bounding_shape = lower_right_top - upper_left_bottom + 1
+
+    d_lim, r_lim, c_lim = np.ogrid[:float(bounding_shape[0]),
+                                   :float(bounding_shape[1]),
+                                   :float(bounding_shape[2])]
+    d_org, r_org, c_org = scaled_center - upper_left_bottom
+    d_rad, r_rad, c_rad = axis_lengths
+    conversion_matrix = R.T @ np.diag(spacing)
```

Comment:
And why is the inverse of the rotation used here?

Reply:
Probably because of the above.

Reply:
I woke up last night, realizing that I had lied to you. It gets more complicated: scikit-image, in a regrettable but somewhat rational early API decision, chose to adopt the XYZ coordinate system for transformations. So that's layered on top of everything. We'll fix that in v2.

Reply:
> I woke up last night, realizing that I had lied to you.

Good to know this doesn't only happen to me!

Reply:
Here's an idea, for a future implementation.

What about a new `scikit.Rotation` class that inherits from `scipy.spatial.transform.Rotation` class, but only changes the `apply` method in order to invert which axes to use (XYZ or ZXY), and `as_euler`/`from_euler` to use your preferred standard of `0,1,2` instead of `X,Y,Z`?

This shouldn't be too time-consuming to implement with inheritance and would allow everyone to use whatever they prefer using (matrices, quaternions or angles).